intro-to-git/commits.md

```diff
@@ -0,0 +1,290 @@
+# Commits
+
+## Learning Goals
+
+- Explain how a commit in Git is made using the git add and git commit commands
+- Practice versioning code by creating commits in Git
+- Practice observing commit history using the commands git status, git log, git show, and git diff
+- Explain different ways of "undoing" commits using git revert and git reset
+
+## Introduction
+
+We know that Git will help us do these two things:
+
+- Version our software with commits and look at our commit history
+- Collaborate with others
+
+The foundation of Git is understanding commits. A solid understanding of commits allows programmers to see their project's progress.
+
+Sometimes, we work on a project alone. Git is still an incredibly powerful and helpful tool, even without the collaboration!
+
+## Vocabulary and Synonyms
+
+| Vocab | Definition | Synonyms | How to Use in a Sentence |
+| ----- | ---------- | -------- | ------------------------ |
+
+Code Change
+Commit
+Local Changes Area
+Staging Area
+Untracked Area
+Tracking
+vim
+
+## Format
+
+The goal of this lesson is to learn how to build commits in Git. However, there is plenty of information to consider before that.
+
+This lesson is organized into three sections: We'll learn about...
+
+1. more details about what commits are
+1. the overall process of making a commit
+1. the steps to take to make a commit
+
+## How Do We See Differences in Code?
+
+One piece of information that commits contain is the differences (diff) in code between this version, and the last version.
+
+We can generalize that each change in software is a set of files and...
+
+- The lines of code that were added; what they are, what file, and what line number
+- The lines of code that were deleted
+- The lines of code that were modified
+
+Let's look at an example.
+
+Nakita is working on a project named Hike Planner, that helps plan her hiking trips.
+
+Nakita starts with this code...
+
+```python
+def convert_to_fahrenheit(deg_celsius):
+    result = (deg_celsius * 1.8) + 32
+    return result
+```
+
+And at the end of the day, she's refactored the function to this, which removes the variable `result`:
+
+```python
+def convert_to_fahrenheit(deg_celsius):
+    return (deg_celsius * 1.8) + 32
+```
+
+We can describe the diff between these two like this:
+
+- remove the line `result = (deg_celsius * 1.8) + 32`
+- remove the line `return result`
+- add the line `return (deg_celsius * 1.8) + 32`
+
+The diff of these changes can be displayed like this:
+
+```
+ def convert_to_fahrenheit(deg_celsius):
+-    result = (deg_celsius * 1.8) + 32
+-    return result
++    return (deg_celsius * 1.8) + 32
+```
+
+<!-- Include code screenshot in order to show color -->
```

Comment:
<img width="763" alt="local screenshot of the diff" src="https://user-images.githubusercontent.com/16619004/103398271-d882f680-4af0-11eb-9f69-9a4af1df1d0b.png">